enrolments/validators.py

```diff
@@ -0,0 +1,110 @@
+from datetime import datetime
+from django.core.exceptions import ValidationError
+from registration.models import Student
+
+
+def validate_attendance(classObj):
+    """
+    Validates against the attendance column for the Class model.
+    The expected structure is:
+    [
+        {
+            "date": "2021-04-19",
+            "attendees": [
+                // student IDs
+                1, 2, 3, 4, 5
+            ]
+        },
+        {
+            "date": "2021-04-26",
+            "attendees": [
+                // student IDs
+                1, 2, 3, 4, 5
+            ]
+        }
+    ]
+    """
+
+    schema = [{"date": "str", "attendees": ["int"]}]
+    if not validate_schema(classObj, schema):
+        raise ValidationError("invalid json structure", code="invalid_schema")
+    for session in classObj:
+        date = session["date"]
+        attendees = session["attendees"]
+        try:
+            datetime.strptime(date, "%Y-%m-%d").date()
+        except ValueError:
+            raise ValidationError(
+                date + " must be formatted as YYYY-MM-DD and must be a valid date",
+                code="invalid_date",
+            )
+        if attendees and not set(
+            Student.objects.filter(pk__in=attendees).values_list("id", flat=True)
+        ) == set(attendees):
+            raise ValidationError(
+                "one or more of the following attendee IDs do not exist: "
+                + str(attendees),
+                code="invalid_attendee",
+            )
+
+
+def validate_schema(json, schema, strict=False):
+    """
```

Comment:
Is there a doc string template we should be using across the board? I'm not sure if there's a standardized one.

Reply:
Do you mean for the schema? The one included in the comments works! I can write out a better-fleshed doc if that helps